apps/demo-typegen/src/demo.test.ts

```diff
@@ -9,7 +9,7 @@ import bytecode from './generated-types/DemoContractAbi.hex';
 
 describe('ExampleContract', () => {
   it('should return the input', async () => {
-    const provider = new Provider('http://127.0.0.1:4000/graphql');
+    const provider = await Provider.connect('http://127.0.0.1:4000/graphql');
```

Comment:
For end-users, this is the _big_ breaking change that this PR will introduce.

Reply:
@camsjams I actually tried what you are proposing in #1173, and while I did make it work for contracts - it is a very hacky solution and I could not make it work for script calls.

After having tried that, I found it easier to just fetch these consensus parameters and pass them down instead of doing it later on and trying to make a lot of things that are sync right now async. This is also what the Rust SDK does.

I agree that us having to rewrite docs, and users having to rewrite their code is a bit of a hassle but I believe it might be worth doing it now rather than later on.